### PR TITLE
Components: fix incorrect style of Guide component on mobile devices

### PR DESCRIPTION
**Pull request by arthur791004**
## Description

The Guide UI is broken on mobile devices, so update the styles to fix the problem

## How has this been tested?

1. Start storybook
2. Select Guide > default
3. Change the size of the preview to **Small mobile (P)**

## Types of changes

Bug fix

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: packages/components/src/guide/style.scss
Comment (stokesman) [inline packages/components/src/guide/style.scss]:
```suggestion
		margin: auto;
		min-width: 280px;
		max-width: calc(100vw - #{$grid-unit-20} * 2);
		
		 // This can be avoided if we add styles to let the image resize or don't change the min-width
		&.components-modal__content {
			overflow-x: hidden;
		}
```
Using auto margins instead of positioning is simpler and goes along with the style refactor from #31639. Removal of `position: absolute` means the footer rules needs an accompanying change but as mentioned in my other comment I think that is already wanting.

**Review by stokesman — COMMENTED** [review]
Hi @arthur791004, thanks for finding and addressing this! (It appears the issue arose due to changes I made in #31639 🤦 ). Testing this out, I came across a couple of things that can be seen in this screenshot:
image

There can be horizontal scrolling (due to the image not resizing) and the next and previous buttons are able to overlay the guide content. The latter is apparently an existing issue and the former is due to the reduced `min-width` added in this PR.

I think I agree with the reduced `min-width`. I'd even favor no minimum width if we were to add style to make the image resize (it would want more than just `max-width: 100%`). As is, I think we can get away with hiding the overflow on the x-axis (though the graphic does get a bit off center). Alternatively, we could leave any `min-width` change out of this PR.

I'd like if we address the footer overlap in this PR too. It should do to just remove the position styles from `components-guide__footer`. That or adding styles to make the footer work without issue as an absolutely positioned element. The latter seems to have been the intent but I don't think the former has any objectionable effect and it's simpler.

**Comment by arthur791004** [comment]
@stokesman I've removed absolute positioning and also disabled the horizontal scrolling!

File: packages/components/src/guide/style.scss
Comment (stokesman) [inline packages/components/src/guide/style.scss]: After testing a bit more I'm favoring omitting the change to `min-width`. The component fits at 320px with a little room on each side and even down at 280px the "next" and "get started" buttons are still accessible. I think the reduced min-width is good but seems non-crucial and better in a separate PR which also styles the images to resize instead of just hiding their overflow.
```suggestion
```

File: packages/components/src/guide/style.scss
Comment (stokesman) [inline packages/components/src/guide/style.scss]: My previous suggestion was botched. The `&` needed a space after it or to not be there at all. Besides, if we were to use this it would go better I think in the ruleset at line 9.
```suggestion
```

**Comment by stokesman** [comment]
@arthur791004, let me know if I've missed the reason we should (in this PR) change the `min-width`. Other than that, I think this is good to go.

**Comment by arthur791004** [comment]
@stokesman No, I just want to keep left and right margin at least `$grid-unit-20` in the small screen (320px). If max-width is smaller than 320px, then the gap might be very small. I'll remove it.

**Review by stokesman — APPROVED** [review]
Great! To sum it up, this prevents the paging buttons from straying outside their parent and from overlaying the guide content in case of short viewports. Thanks again @arthur791004 🚀

**Comment by arthur791004** [comment]
Thanks @stokesman too 🙂

**Comment by arthur791004** [comment]
@stokesman Could you help to merge this? I don't have permission for it 😞 Thanks!

**Comment by stokesman** [comment]
@arthur791004, Thanks for the reminder on this. By [the books](https://github.com/WordPress/gutenberg/blob/trunk/docs/contributors/repository-management.md#merging-pull-requests) we're supposed to have a core member’s approval before merging. I've requested a review from @jasmussen.

**Review by jasmussen — APPROVED** [review]
This is testing well for me! I still do see a vertical scrollbar on this screen, but I doubt we can entirely eliminate such, at least there isn't a horizontal scrollbar.

Screenshot 2021-08-10 at 10 35 26

Thanks for the PR, let's land it.

> Thanks for the reminder on this. By the books we're supposed to have a core member’s approval before merging. I've requested a review from @jasmussen.

At this rate @stokesman I see no reason why you wouldn't count as a core member. Appreciate you going by the books, but in case you'd be interested, I doubt anyone would object to granting you core member status. You need only ask in #core-editor. Thanks for all your contributions!

**Pull request merged**
